src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -0,0 +1,268 @@
+        adTaskManager
+            .updateLatestRealtimeTaskOnCoordinatingNode(
+                detectorId,
+                taskState,
+                rcfTotalUpdates,
+                detectorIntervalInMinutes,
+                error,
+                ActionListener.wrap(r -> {
+                    if (r != null) {
+                        log.debug("Updated latest realtime task successfully for detector {}, taskState: {}", detectorId, taskState);
+                    }
+                }, e -> {
+                    if ((e instanceof ResourceNotFoundException) && e.getMessage().contains(CAN_NOT_FIND_LATEST_TASK)) {
+                        // Clear realtime task cache, will recreate AD task in next run, check AnomalyResultTransportAction.
+                        log.error("Can't find latest realtime task of detector " + detectorId);
+                        adTaskManager.removeRealtimeTaskCache(detectorId);
+                    } else {
+                        log.error("Failed to update latest realtime task for detector " + detectorId, e);
+                    }
+                })
+            );
+    }
+
+    public void indexAnomalyResultException(
+        Instant detectionStartTime,
+        Instant executionStartTime,
+        String errorMessage,
+        String taskState,
+        AnomalyDetector detector
+    ) {
+        String detectorId = detector.getDetectorId();
+        try {
+            IntervalTimeConfiguration windowDelay = (IntervalTimeConfiguration) detector.getWindowDelay();
+            Instant dataStartTime = detectionStartTime.minus(windowDelay.getInterval(), windowDelay.getUnit());
+            Instant dataEndTime = executionStartTime.minus(windowDelay.getInterval(), windowDelay.getUnit());
+            User user = detector.getUser();
+
+            AnomalyResult anomalyResult = new AnomalyResult(
+                detectorId,
+                null, // no task id
+                new ArrayList<FeatureData>(),
+                dataStartTime,
+                dataEndTime,
+                executionStartTime,
+                Instant.now(),
+                errorMessage,
+                null, // single-stream detectors have no entity
+                user,
+                anomalyDetectionIndices.getSchemaVersion(ADIndex.RESULT),
+                null // no model id
+            );
+            String resultIndex = detector.getResultIndex();
+            if (resultIndex != null && !anomalyDetectionIndices.doesIndexExist(resultIndex)) {
+                // Set result index as null, will write exception to default result index.
+                anomalyResultHandler.index(anomalyResult, detectorId, null);
+            } else {
+                anomalyResultHandler.index(anomalyResult, detectorId, resultIndex);
+            }
+
+            if (errorMessage.contains(CommonErrorMessages.NO_CHECKPOINT_ERR_MSG) && !detector.isMultiCategoryDetector()) {
```

Comment:
Do single entity detectors also need 1 minute? if we restarted the detector is there any reason for us to believe initialization will be any quicker in the case we already have some results or when stopping detector we delete the model checkpoint along with any other relevant stored info?

Reply:
yeah, 1 minute is to give enough leeway for training and checkpoint write, regardless of single stream or HCAD detectors.

For restart detector, as mentioned earlier, plan to do it after the speed up PR is merged so the PR is not too big. Currently, it is still an issue.

When stopping detector, we will delete everything except results. It will have to go through a new round of cold start.

Reply:
sounds good

Reply:
For the restart scenario, since the checkpoint would have been deleted already, wouldn't it automatically go through this flow and have the 1 minute delay? In other words, what needs to be done differently in the restart scenario vs. the initial start scenario?

I may answer my own question as I keep reading

Reply:
note it is restarting cluster not detector. So the checkpoint won't be deleted.

Reply:
it is a typo. Will change to cluster restart.